### PR TITLE
Question & Answer CRUD

### PR DESCRIPTION
**Pull request by KavikaPalletenne**

### CODE REVIEW

File: backend/server/src/handler/rating.rs [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+pub struct RatingHandler;
 
-impl RatingsHandler {
+impl RatingHandler {
     // TODO: are all the user permissions as required? Who should be able to do what with ratings?
-    pub async fn create_rating(
+    pub async fn create(
         State(state): State<AppState>,
         Path(application_id): Path<i64>,
         _admin: ApplicationCreatorGivenApplicationId,
```
Comment (AlexMIaoPU) [inline backend/server/src/handler/rating.rs:20]: Didn't catch this earlier but this naming should be RatingCreatorGivenApplicationId

Comment (AlexMIaoPU) [inline backend/server/src/handler/rating.rs:20]:
```suggestion
        _admin: ApplicationCreatorGivenApplicationId,
```

Didn't catch this earlier but this naming should be RatingCreatorGivenApplicationId

File: backend/server/src/handler/rating.rs [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 use axum::extract::{Json, Path, State};
 use axum::http::StatusCode;
 use axum::response::IntoResponse;
 
-pub struct RatingsHandler;
+pub struct RatingHandler;
 
-impl RatingsHandler {
+impl RatingHandler {
     // TODO: are all the user permissions as required? Who should be able to do what with ratings?
```
Comment (AlexMIaoPU) [inline backend/server/src/handler/rating.rs:16]:
```suggestion
```

File: backend/server/src/models/answer.rs [hunk 3]
```diff
@@ -32,66 +35,417 @@ pub struct Answer {
...
+}
+
+#[derive(Deserialize, sqlx::FromRow)]
+pub struct AnswerRawData {
+    id: i64,
+    question_id: i64,
+    question_type: QuestionType,
+    short_answer_answer: Option<String>,
+    multi_option_answers: Option<Vec<i64>>,
+    ranking_answers: Option<Vec<i64>>,
```
[the remark this answers is not in the document]
Comment (KavikaPalletenne) [reply backend/server/src/models/answer.rs:54]: It's used to keep it as one db query. Not sure if object enum work with sqlx.

File: backend/server/src/models/organisation.rs [hunk 4]
```diff
@@ -175,6 +175,13 @@ impl Organisation {
         admin_id_list: Vec<i64>,
         transaction: &mut Transaction<'_, Postgres>,
     ) -> Result<(), ChaosError> {
+        let _ = sqlx::query!(
```
[the remark this answers is not in the document]
Comment (KavikaPalletenne) [reply backend/server/src/models/organisation.rs:178]: No `organisation_members` rows for a given `organisation_id` does not mean the organisation doesn't exist, so we need to check it exists first

**Pull request merged**